ergo-wallet/src/main/scala/org/ergoplatform/wallet/secrets/DerivationPath.scala

```diff
@@ -57,6 +61,32 @@ object DerivationPath {
     }
   }
 
+  /**
+    * Finds next available path index for a new key.
+    */
+  def nextPath(secrets: IndexedSeq[ExtendedSecretKey]): Try[DerivationPath] = {
+    @scala.annotation.tailrec
+    def nextPath(accPath: List[Int], rem: Seq[Seq[Int]]): Try[DerivationPath] = {
+      if (!rem.forall(_.isEmpty)) {
+        val maxChildIdx = rem.flatMap(_.headOption).max
+        if (!Index.isHardened(maxChildIdx)) {
+          Success(DerivationPath(0 +: (accPath :+ maxChildIdx + 1), publicBranch = false))
+        } else {
+          nextPath(accPath :+ maxChildIdx, rem.map(_.drop(1)))
+        }
+      } else {
+        Failure(
+          new Exception("Out of non-hardened index space. Try to derive hardened key specifying path manually"))
+      }
+    }
+
+    if (secrets.size == 1) {
+      Success(DerivationPath(Array(0, 1), publicBranch = false))
+    } else {
+      nextPath(List.empty, secrets.map(_.path.decodedPath.tail.toList))
```

Comment:
please use local variable to give `_.path.decodedPath.tail.toList` some meaningful name.

Reply:
Made named closure, as well as this PR https://github.com/ergoplatform/ergo/issues/1082